### PR TITLE
[Ep-458] QA Page Viewed (Project) Phase 1

### PR DESCRIPTION
**Pull request by tobitech**
# 📲 What

Page viewed for project was tracked in two places. 
Moved the one for campaign section context into Project Description screen when the content loads.
Add RefTag as part of the Config Data sent to ProjectDescription screen.

# 🤔 Why

This is part of phase 1 Events QA.

# 👀 See

iOS-QA-Page-Viewed-(Project)-Phase-1

### CODE REVIEW

File: Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift [hunk 1]
```diff
@@ -554,8 +554,8 @@ public final class AppDelegateViewModel: AppDelegateViewModelType, AppDelegateVi
 
     let campaignFaqLink = projectLink
       .filter { _, subpage, _, _ in subpage == .faqs }
-      .map { project, _, vcs, _ in
-        vcs + [ProjectDescriptionViewController.configuredWith(value: project)]
+      .map { project, _, vcs, refTag in
+        vcs + [ProjectDescriptionViewController.configuredWith(data: (project, refTag))]
```
Comment (singhhari) [inline Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift:558]: Can you address the coverage issue here?

File: Library/ViewModels/ProjectDescriptionViewModel.swift [hunk 2]
```diff
@@ -5,7 +5,7 @@ import WebKit
 
 public protocol ProjectDescriptionViewModelInputs {
   /// Call with the project given to the view.
-  func configureWith(value: Project)
+  func configureWith(value: ProjectPamphletMainCellData)
```
Comment (singhhari) [inline Library/ViewModels/ProjectDescriptionViewModel.swift:8]: why was this changed from project to data?

**Review by singhhari — CHANGES_REQUESTED** [review]
Functionality is good on the Segment debugger but coverage is down a bit

File: Library/ViewModels/ProjectDescriptionViewModel.swift (hunk 2 above)
Comment (tobitech) [reply Library/ViewModels/ProjectDescriptionViewModel.swift:8]: `ProjectPamphletMainCellData` is a tuple of Project and RefTag. Just to make sure that RefTag is sent along when tracking the Project Page Viewed event in that ViewModel.

File: Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift [hunk 3]
```diff
@@ -623,44 +629,16 @@ final class ProjectPamphletMainCellViewModelTests: TestCase {
...
-      )
-      XCTAssertEqual(self.segmentTrackingClient.properties(forKey: "project_subcategory"), ["Art"])
-      XCTAssertEqual(self.segmentTrackingClient.properties(forKey: "project_category"), [nil])
-      XCTAssertEqual(self.segmentTrackingClient.properties(forKey: "project_country"), ["US"])
-      XCTAssertEqual(
-        self.segmentTrackingClient.properties(forKey: "project_user_has_watched", as: Bool.self),
-        [nil]
-      )
-    }
+private func assertProjectRefTagEqual(
```
Comment (singhhari) [inline Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift:634]: Can this be replaced with a standard assertion like we use everywhere else?

File: Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift [hunk 4]
```diff
@@ -623,44 +629,16 @@ final class ProjectPamphletMainCellViewModelTests: TestCase {
       XCTAssertEqual(self.optimizelyClient.trackedAttributes?["session_device_type"] as? String, "phone")
     }
   }
+}
 
-  func testTrackingProjectViewed_LiveProject_LoggedIn_NonBacked() {
```
Comment (singhhari) [inline Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift:627]: Why was this one deleted?

**Review by singhhari — CHANGES_REQUESTED** [review]

File: Kickstarter-iOS/ViewModels/AppDelegateViewModelTests.swift [hunk 5]
```diff
@@ -642,6 +642,16 @@ final class AppDelegateViewModelTests: TestCase {
       XCTAssertTrue(result)
 
       self.presentViewController.assertValues([1, 2, 2, 2, 3])
+
+      let faqUrl = projectUrl + "/faqs"
```
[the remark this answers is not in the document]
Comment (tobitech) [reply Kickstarter-iOS/ViewModels/AppDelegateViewModelTests.swift:646]: Yes it's unrelated.

File: Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift (hunk 3 above)
Comment (tobitech) [reply Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift:634]: `XCTAssertEqual` doesn't work on tuples because they don't conform to `Equatable` protocol that's why I had to do that. Another option would be to pass the `Project` and `RefTag` as separate parameters to the `config()` method of `ProjectDescriptionViewController`

File: Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift (hunk 4 above)
Comment (tobitech) [reply Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift:627]: Because Project Page Viewed event for campaign context was moved away from the `ProjectPamphletMainCellViewModel`.

**Review by singhhari — APPROVED** [review]
lgtm
